### PR TITLE
[RND-628] Switch to pino logger, either console or file logging not both

### PR DESCRIPTION
**Pull request by bradbanister**

### CODE REVIEW

File: Meadowlark-js/services/meadowlark-fastify/.env.example [hunk 1]
```diff
@@ -61,12 +61,12 @@ FASTIFY_PORT=3000
...
-# Valid levels include: error, warning, info, debug
+# Valid levels include: error, warning, info, debug (higher levels provide faster logging performance)
 LOG_LEVEL=debug
-# When true, logs to plain text instead of JSON
+# When true, logs to plain text instead of JSON (slower logging performance)
 IS_LOCAL=true
-# By default, it will not save to a log file, set to true if desired
-# SAVE_LOG_TO_FILE=false
+# By default, it writes logs to STDOUT. Set to true to log to a file instead (faster logging performance)
+# LOG_TO_FILE=false
```
Comment (bradbanister) [inline Meadowlark-js/services/meadowlark-fastify/.env.example:69]: Changed `SAVE_LOG_TO_FILE` to `LOG_TO_FILE` to be more in line with `LOG_LEVEL` and `LOG_FILE_LOCATION`

File: Meadowlark-js/packages/meadowlark-authz-server/src/handler/RequestToken.ts [hunk 2]
```diff
@@ -138,7 +138,9 @@ export async function requestToken(authorizationRequest: AuthorizationRequest):
 
     const { requestTokenBody } = parsedRequest;
 
-    Logger.debug(`${moduleName}.requestToken ${maskClientSecret(requestTokenBody)}`, authorizationRequest.traceId);
+    if (isDebugEnabled()) {
```
Comment (bradbanister) [inline Meadowlark-js/packages/meadowlark-authz-server/src/handler/RequestToken.ts:141]: Guarding debug statements that evaluate functions before the call

File: Meadowlark-js/packages/meadowlark-utilities/src/Logger.ts [hunk 3]
```diff
@@ -40,79 +39,78 @@ const convertErrorToString = (err) => {
...
-let logger: winston.Logger = winston.createLogger({
-  transports: [transport],
-});
-
 /**
  * This should be called by frontend services at startup, before logging. Because services can have
  * multiple startup points (e.g. multiple lambdas), this checks if initialization has already happened.
+ *
+ * Note that writing logs to STDOUT is slower than writing to a file, and pretty-printing with IS_LOCAL
+ * is slower than not.
```
[the remark this answers is not in the document]
Comment (bradbanister) [reply Meadowlark-js/packages/meadowlark-utilities/src/Logger.ts:53]: I didn't look at running in a container. I'd think it depends on whether STDOUT is actually going to a console or not. If redirected to a file it should be fine.

Will add notes.

File: Meadowlark-js/packages/meadowlark-utilities/src/Logger.ts [hunk 4]
```diff
@@ -130,10 +128,14 @@ export const Logger = {
...
   },
   info: (message: string, traceId: string | null, extra?: any | null) => {
-    logger.info({ message: removeSensitiveData(message), traceId, extra });
+    if (isInfoEnabled()) {
+      logger.info({ message: removeSensitiveData(message), traceId, extra });
+    }
   },
   debug: (message: string, traceId: string | null, extra?: any | null) => {
-    logger.debug({ message: removeSensitiveData(message), traceId, extra });
+    if (isDebugEnabled()) {
```
[the remark this answers is not in the document]
Comment (bradbanister) [reply Meadowlark-js/packages/meadowlark-utilities/src/Logger.ts:136]: Yes, we don't want those higher up functions to be called either.

**Review by stephenfuqua — APPROVED** [review]

**Pull request merged**